File: src/vtk_named_colors.rs
#[cxx::bridge]
mod ffi {
    unsafe extern "C++" {
        include!("vtk_named_colors.h");

        type vtkNamedColors;

        fn named_colors_new() -> *mut vtkNamedColors;
        unsafe fn named_colors_delete(named_colors: *mut vtkNamedColors);
        // unsafe fn named_colors_set_input(pdm: *mut vtkNamedColors, data: vtkPolyData);
    }
}

crate::define_object!(
    "https://vtk.org/doc/nightly/html/classvtkNamedColors.html",
    @name NamedColors, *mut ffi::vtkNamedColors,
    @new ffi::named_colors_new,
    @delete ffi::named_colors_delete
);

crate::inherit!(NamedColors vtkAlgorithm);

File: vtk-rs/libvtkrs/src/vtk_named_colors.cpp
#include "vtk_named_colors.h"
#include "cxx.h"
#include "vtk_named_colors.rs.h"

#include <vtkPolyData.h>

vtkNamedColors* named_colors_new() {
    return vtkNamedColors::New();
}

void named_colors_delete(vtkNamedColors& named_colors) {
    named_colors.Delete();
}
